### ISSUE TITLE
Colon spacing violation not triggered in closure declaration

### ISSUE THREAD

**Issue by kgellci** [realm/SwiftLint#2008]
### New Issue Checklist

- [✅ ] Updated SwiftLint to the latest version
- [✅] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Bug Report

Trailing space on colon violation should be triggered when declaring a closure property.

Expected:
❌   `private var action:(() -> Void)?`
✅   `private var action: (() -> Void)?`

Actual:
✅    `private var action:(() -> Void)?`
✅   `private var action: (() -> Void)?`

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
echo "private var action:(() -> Void)?" | swiftlint lint --use-stdin  --no-cache --enable-all-rules
Done linting! Found 0 violations, 0 serious in 1 file.
```

### Environment

* SwiftLint version: 0.24.2
* Installation method used: Homebrew
* Xcode Version: 9.2 (9C40b)

```swift
// This should trigger a violation:
private var action:(() -> Void)?
```

**Comment by keith** [comment]
Still relevant
